src/components/Modal/ModalHeading/ModalHeading.tsx

```diff
@@ -1,11 +1,20 @@
 import React from 'react'
 import classnames from 'classnames'
 
+interface ModalHeadingProps {
+  children?: React.ReactNode | string
+  className?: string
+}
+
 export const ModalHeading = ({
-  className,
   children,
+  className,
   ...headingProps
-}: React.HTMLProps<HTMLHeadingElement>): React.ReactElement => {
+}: ModalHeadingProps &
+  React.DetailedHTMLProps<
+    React.HTMLAttributes<HTMLHeadingElement>,
+    HTMLHeadingElement
+  >): React.ReactElement => {
```

Comment:
Updated this type to be in line with what we were using on `ProcessListHeading`, which the the result of `JSX.IntrinsicElements['h{1-6}']`

@suzubara is that an okay change?